### PR TITLE
Wrap url.Error in APIError

### PR DESCRIPTION
**Pull request by mgyucht**
## Changes
https://github.com/databricks/databricks-sdk-go/commit/b0f3c83d9f5eca12dbd4c1daa03e06aff98a94ff introduced a regression in the handling of errors returned from HttpClient.Do. Previously, only some errors returned by Client.Do() were retried; now, all are retried. This PR adds default error handling logic in httpclient for errors returned by Client.Do() matching the previous behavior. As part of this, we remove the duplicated transient error messages between apierr and the default client.

## Tests

- [ ] `make test` passing
- [ ] `make fmt` applied
- [ ] relevant integration tests applied

### CODE REVIEW

**Review by pietern — COMMENTED** [review]
Discussed in person and decided on treating `url.Error` retries separately from `apierr.APIError` retries, given the former are for lower level networking issues and the latter for when we actually got a response from the server.

File: client/client.go [hunk 1]
```diff
@@ -65,16 +65,6 @@ func New(cfg *config.Config) (*DatabricksClient, error) {
 			},
 			TransientErrors: []string{
 				"REQUEST_LIMIT_EXCEEDED", // This is temporary workaround for SCIM API returning 500.  Remove when it's fixed
```
Comment (pietern) [inline client/client.go:67]: Looking at this sole example remaining, the field could be removed and folded into `ErrorRetriable` as well. In this case in the set of transient errors in `apierr` because it is SCIM related (can be another PR).

File: httpclient/api_client_test.go [hunk 2]
```diff
@@ -140,6 +140,41 @@ func TestSimpleRequestRetried(t *testing.T) {
...
+	})
+	var resp Dummy
+	err := c.Do(context.Background(), "PATCH", "/a",
+		WithRequestData(Dummy{1}),
+		WithResponseUnmarshal(&resp))
+	require.Error(t, err)
+	require.ErrorIs(t, err, transportErr)
+	require.True(t, tried, "request was not tried")
+	require.False(t, retried, "request was retried")
+}
```
[the remark this answers is not in the document]
Comment (mgyucht) [reply httpclient/api_client_test.go:176]: Yeah, it's `TestSimpleRequestRetried` in httpclient/api_client_test.go
